### PR TITLE
[Enhancement] MMRotate 1.x support

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support MMRotate in dev-1.x

## Modification

- [x] update rewriters
- [x] refactor nms
- [x] documentation
- [x] sdk

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/04-supported-codebases/mmrotate.md [hunk 1]
```diff
@@ -18,16 +19,170 @@ ______________________________________________________________________
...
 
+你可以使用 [tools/deploy.py](https://github.com/open-mmlab/mmdeploy/blob/dev-1.x/tools/deploy.py) 把 mmrotate 模型一键式转换为推理后端模型。
+该工具的详细使用说明请参考[这里](https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/02-how-to-run/convert_model.md#usage).
+
+以下，我们将演示如何把 `rotated-faster-rcnn` 转换为 onnx 模型。
+
+```shell
+cd mmdeploy
+
+# download unet model from mmrotate model zoo
```
Comment (AllentDan) [inline docs/zh_cn/04-supported-codebases/mmrotate.md:63]: unet model?

File: docs/en/04-supported-codebases/mmrotate.md [hunk 2]
```diff
@@ -18,16 +19,166 @@ ______________________________________________________________________
...
+You can also perform SDK model inference like following,
+
+```python
+from mmdeploy_python import RotatedDetector
+import cv2
+import numpy as np
+
+img = cv2.imread('./dota_demo.jpg')
+
+# create a classifier
```
Comment (AllentDan) [inline docs/en/04-supported-codebases/mmrotate.md:169]: detector

File: mmdeploy/backend/sdk/export_info.py [hunk 3]
```diff
@@ -159,6 +159,18 @@ def get_preprocess(deploy_cfg: mmengine.Config, model_cfg: mmengine.Config,
...
+    # remove scope name
+    # NOTE: There is a potential risk that transforms with same names in
+    # different scope might have different behavior
+    for transform in transforms:
+        trans_type = transform['type']
+        trans_type = trans_type.split('.')
+        if len(trans_type) > 1:
+            trans_type = trans_type[1:]
+        trans_type = '.'.join(trans_type)
+        transform['type'] = trans_type
```
Comment (AllentDan) [inline mmdeploy/backend/sdk/export_info.py:172]: If the part only works for MMRotate, we may consider moving it to `get_preprocess` of mmrotate.

File: tests/test_codebase/test_mmdet/test_mmdet_models.py [hunk 4]
```diff
@@ -877,10 +778,12 @@ def _replace_r50_with_r18(model):
...
 
 @pytest.mark.parametrize('backend', [Backend.ONNXRUNTIME])
-@pytest.mark.parametrize('model_cfg_path', [
-    'tests/test_codebase/test_mmdet/data/single_stage_model.json',
-    'tests/test_codebase/test_mmdet/data/mask_model.json'
-])
+@pytest.mark.parametrize(
+    'model_cfg_path',
+    [
+        # 'tests/test_codebase/test_mmdet/data/single_stage_model.json',
```
Comment (AllentDan) [inline tests/test_codebase/test_mmdet/test_mmdet_models.py:784]: may remove if useless

File: mmdeploy/backend/sdk/export_info.py (hunk 3 above)
Comment (grimoire) [reply mmdeploy/backend/sdk/export_info.py:172]: I am not sure. This would happen when a repo reuse modules/transforms in another repo.

Comment (AllentDan) [reply mmdeploy/backend/sdk/export_info.py:172]: By far, only mmrotate shows the attribute. Maybe MMOCR in the future.

**Comment by RunningLeon** [comment]
@grimoire may need to fix conflicts

File: demo/python/rotated_object_detection.py [hunk 5]
```diff
@@ -22,8 +22,7 @@ def main():
     args = parse_args()
 
     img = cv2.imread(args.image_path)
-    detector = RotatedDetector(
-        model_path=args.model_path, device_name=args.device_name, device_id=0)
+    detector = RotatedDetector(args.model_path, args.device_name, 0)
```
Comment (RunningLeon) [inline demo/python/rotated_object_detection.py:25]: pls update regression yml config and do the test.
https://github.com/grimoire/mmdeploy/blob/mmrotate-1.x-support/tests/regression/mmrotate.yml

File: tests/regression/mmrotate.yml [hunk 6]
```diff
@@ -49,7 +49,7 @@ models:
   - name: oriented_rcnn
     metafile: configs/oriented_rcnn/metafile.yml
     model_configs:
-      - configs/oriented_rcnn/oriented_rcnn_r50_fpn_fp16_1x_dota_le90.py
+      - configs/oriented_rcnn/oriented-rcnn-le90_r50_fpn_1x_dota.py
```
Comment (RunningLeon) [inline tests/regression/mmrotate.yml:52]: ort failed
```
  File "/opt/conda/envs/torch1.8.0/lib/python3.8/site-packages/mmdet/models/dense_heads/base_dense_head.py", line 197, in predict
    predictions = self.predict_by_feat(
  File "/root/workspace/mmdeploy/mmdeploy/core/rewriters/rewriter_utils.py", line 379, in wrapper
    return self.func(self, *args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/codebase/mmrotate/models/dense_heads/oriented_rpn_head.py", line 168, in rpn_head__predict_by_feat
    dets, labels, index = multiclass_nms(
  File "/root/workspace/mmdeploy/mmdeploy/core/optimizers/function_marker.py", line 261, in g
    rets = f(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/mmcv/ops/nms.py", line 513, in multiclass_nms
    return _multiclass_nms(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/core/rewriters/rewriter_utils.py", line 379, in wrapper
    return self.func(self, *args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/mmcv/ops/nms.py", line 432, in multiclass_nms__default
    return ctx.origin_func(
  File "/root/workspace/mmdeploy/mmdeploy/mmcv/ops/nms.py", line 299, in _multiclass_nms
    assert not output_index, 'output_index is not supported on this backend.'
AssertionError: output_index is not supported on this backend.
12/05 07:45:01 - mmengine - ERROR - /root/workspace/mmdeploy/mmdeploy/apis/core/pipeline_manager.py - pop_mp_output - 80 - `mmdeploy.apis.pytorch2onnx.torch2onnx` with Call id: 0 failed. exit.
```

File: tests/test_codebase/test_mmrotate/data/model.py [hunk 7]
```diff
@@ -1,96 +1,240 @@
...
+    dict(type='mmdet.LoadAnnotations', with_bbox=True, box_type='qbox'),
+    dict(type='ConvertBoxType', box_type_mapping=dict(gt_bboxes='rbox')),
+    dict(type='mmdet.Resize', scale=(1024, 1024), keep_ratio=True),
+    dict(
+        type='mmdet.RandomFlip',
+        prob=0.75,
+        direction=['horizontal', 'vertical', 'diagonal']),
+    dict(type='mmdet.PackDetInputs')
+]
+val_pipeline = [
```
Comment (AllentDan) [inline tests/test_codebase/test_mmrotate/data/model.py:18]: Seems useless. May remove redundant content.

**Review by AllentDan — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**